zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/DataSetInfo.scala

```diff
@@ -13,3 +18,120 @@ case class DataSetInfo(name: String,
                        schema: Schema,
                        dataInterval: Interval,
                        stats: Stats)
+
+object DataSetInfo {
+
+  def parse(json: JsValue): DataSetInfo = {
+    json.validate[DataSetInfo] match {
+      case js: JsSuccess[DataSetInfo] => js.get
+      case e: JsError => throw JsonRequestException(JsError.toJson(e).toString())
+    }
+  }
+
+  def write(dataSetInfo: DataSetInfo): JsValue = Json.toJson(dataSetInfo)
+
+  implicit val intervalFormat: Format[Interval] = new Format[Interval] {
+    override def reads(json: JsValue) = {
+      val start = (json \ "start").as[DateTime]
```

Comment:
to avoid the timezone related parsing problems, usually we just store the `getMillis` as `Long` other than `DateTime`.